mailet/standard/src/main/java/org/apache/james/transport/mailets/AmqpForwardAttribute.java

```diff
@@ -0,0 +1,138 @@
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport.mailets;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Map;
+import java.util.concurrent.TimeoutException;
+
+import org.apache.mailet.Mail;
+import org.apache.mailet.MailetException;
+import org.apache.mailet.base.GenericMailet;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.rabbitmq.client.AMQP;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+
+/**
+ * Forward the attributes values, expected to be a Map&lt;String, byte[]&gt;
+ * to a AMPQ defined by its exchange and its routing key (defaulting to empty string)
+ */
+public class AmqpForwardAttribute extends GenericMailet {
+
+    public static final String URI_PARAMETER_NAME = "uri";
+    public static final String EXCHANGE_PARAMETER_NAME = "exchange";
+    public static final String ROUTING_KEY_PARAMETER_NAME = "routing_key";
+    public static final String ATTRIBUTE_PARAMETER_NAME = "attribute";
+
+    public static final String ROUTING_KEY_DEFAULT_VALUE = "";
+
+    private String exchange;
+    private String attribute;
+    private ConnectionFactory connectionFactory;
+    @VisibleForTesting String routingKey;
+
+    public void init() throws MailetException {
+        String uri = getInitParameter(URI_PARAMETER_NAME);
+        if (uri == null) {
+            throw new MailetException("No value for " + URI_PARAMETER_NAME
+                    + " parameter was provided.");
+        }
+        exchange = getInitParameter(EXCHANGE_PARAMETER_NAME);
+        if (exchange == null) {
+            throw new MailetException("No value for " + EXCHANGE_PARAMETER_NAME
+                    + " parameter was provided.");
+        }
+        routingKey = getInitParameter(ROUTING_KEY_PARAMETER_NAME, ROUTING_KEY_DEFAULT_VALUE);
+        attribute = getInitParameter(ATTRIBUTE_PARAMETER_NAME);
+        if (attribute == null) {
+            throw new MailetException("No value for " + ATTRIBUTE_PARAMETER_NAME
+                    + " parameter was provided.");
+        }
+        connectionFactory = new ConnectionFactory();
+        try {
+            connectionFactory.setUri(uri);
+        } catch (Exception e) {
+            throw new MailetException("Invalid " + URI_PARAMETER_NAME
```

Comment:
```
throw new MailetException("Invalid " + URI_PARAMETER_NAME + " parameter was provided: " + uri, e);
```

Otherwise we loose details about what's happening....